### ISSUE TITLE
Should the Process instances be disposed?

### ISSUE THREAD

**Issue by paolofulgoni** [dotnet/dotnet-api-docs#793]
Please clarify if the `Process` instances returned by the `GetProcessesByName` method should be disposed by the caller.
In that case, please update the examples accordingly.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ad7b0b29-4df3-aefd-2695-c64fa90f3d51
* Version Independent ID: 08bf7fd3-54a6-aad5-95dc-81f4d6819d3e
* Content: [Process.GetProcessesByName Method (System.Diagnostics)](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.getprocessesbyname?view=netframework-4.7.2#System_Diagnostics_Process_GetProcessesByName_System_String_)
* Content Source: [xml/System.Diagnostics/Process.xml](https://github.com/dotnet/dotnet-api-docs/blob/master/xml/System.Diagnostics/Process.xml)
* Product: **.net**
* GitHub Login: @dotnet-bot
* Microsoft Alias: **dotnetcontent**

**Comment by MSDN-WhiteKnight** [comment]
Process instance should be disposed if it owns an open OS process handle. Process instances returned by GetProcessesByName method initially do not open a handle to the process, but they do it when the caller accesses some of their members, such as [Handle](https://referencesource.microsoft.com/System/R/41ed61685806ec61.html) property. 

So, caller should definitely dispose all Process instances that it "touched", i.e. for which it called some properties or methods. Disposing unused Process instances seems unneed. But it might be only the detail of current implementation, so i'm not sure what documentation should recomend on this.

**Comment by paolofulgoni** [comment]
Thanks for the detailed response.
So if disposing is sometimes needed and sometimes not, but never a problem, then disposing all the times seems a good practice to me. What do you think?

**Comment by MSDN-WhiteKnight** [comment]
Yes, disposing all instances is surely a safe choice. And, until someone officially states otherwise, it is assumed a good practice.

**Comment by paolofulgoni** [comment]
Back to my original question, I think that the examples should be updated accordingly
